app/change_set_persisters/hyrax/change_set_persister.rb

```diff
@@ -3,43 +3,79 @@
+    #   apply the changes to the resource in
+    # @param [Valkyrie::StorageAdapter] the storage_adapter you wish to store the
+    #   changes to the resource in
     def initialize(metadata_adapter:, storage_adapter:)
       @metadata_adapter = metadata_adapter
       @storage_adapter = storage_adapter
     end
 
+    # Saves all changes in the ChangeSet to the resource contained in the ChangeSet
+    # using the metadata_adapter and storage_adapter
+    # @param [Hyrax::DynamicChangeSet] change_set The change_set whose resource you wish to save
+    # @return the saved resource, such as GenericWork, that is subclassed from Valkyrie::Resource
     def save(change_set:)
-      result = nil
-      catch(:abort) do
-        run_hook(:before_save, change_set: change_set)
-        result = persister.save(resource: change_set.resource)
-        run_hook(:after_save, change_set: change_set, resource: result)
-      end
-      result
+      run_hook(:before_save, change_set: change_set)
+      resource = persister.save(resource: change_set.resource)
+      run_hook(:after_save, change_set: change_set)
+      query_service.find_by(id: resource.id)
     end
 
+    # Deletes the resource contained in the change_set from the adapters specified
+    # upon initialization of the ChangeSetPersister
+    # @param [Hyrax::DynamicChangeSet] change_set The change_set whose resource you wish to delete
     def delete(change_set:)
-      result = nil
-      catch(:abort) do
-        run_hook(:before_delete, change_set: change_set)
-        result = persister.delete(resource: change_set.resource)
-        run_hook(:after_delete, change_set: change_set)
-      end
-      result
+      run_hook(:before_delete, change_set: change_set)
+      persister.delete(resource: change_set.resource)
+      run_hook(:after_delete, change_set: change_set)
     end
 
+    # Calls save on all change_sets in the array passed in
+    # @param [Array <Hyrax::DynamicChangeSet>] change_sets An array of change_sets whose resources you wish to save
+    # @return [Array <Valkyrie::Resource>] an array of the saved resources, such as GenericWork,
+    #  that is subclassed from Valkyrie::Resource
     def save_all(change_sets:)
       change_sets.map do |change_set|
         save(change_set: change_set)
       end
     end
 
+    # Calls delete on all change_sets in the arraypassed in
+    # @param [Array <Hyrax::DynamicChangeSet>] change_sets An array of change_sets whose resources you wish to delete
+    #  all resources were succesfully deleted
+    def delete_all(change_sets:)
+      change_sets.map do |change_set|
+        delete(change_set: change_set)
+      end
+    end
+
+    # Deletes all file_sets that are members of the resource in the supplied change_set
+    # @param [Hyrax::DynamicChangeSet] the change_set that contains the resource whose member file_sets you wish to delete
+    def cleanup_file_sets(change_set:)
```

Comment:
Should this be called by some code (other than tests?) How about moving to a WorkChangeSetPersister?